lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -3032,8 +3041,19 @@ public function registerDocument($document, $id)
         $oid = spl_object_hash($document);
         $this->documentIds[$oid] = $id;
         $this->identityMap[$id] = $document;
-        $this->setDocumentState($oid, self::STATE_MANAGED);
-
+        
+        // frozen nodes need another state so they are managed but not included for updates
+        $nodeType = '';
+        try {
+        	$node = $this->session->getNode($id);
+        	// In some of the unit tests $node will be null because of how they are mocked
+        	$nodeType = $node ? $node->getPrimaryNodeType()->getName() : '';
```

Comment:
lets do something like this:

```
$frozen = $this->session->hasNode($id) && $this->session->getNode($id)->isNodeType('nt:frozen');
```

alternatively, you could add an optional argument to pass the state along. add a new hint to getOrCreateDocuments and then pass the value of that hint to registerDocument. then you do not need to check for node types in the registerDocument method.